### DISCUSSION TITLE
Dockerfiles für open3e

### DISCUSSION THREAD

**Discussion by flecke-m** [open3e/open3e#110] [Show and tell]
Hallo zusammen,

erstmal Danke für diese tolle Projekt, da ich auch ein Viessmann Leidensgenosse bin, benötige ich dringend etwas um die Geräte lokal auszulesen und in Homeassistant zu verarbeiten.

Da ich ein absoluter Fan von Containern bin, habe ich mal zwei Dockerfiles erstellt (einen für depictSystem und einen für open3e). Diese basieren auf dem offiziellen Python - Alpine Image und sind jeweils knapp unter 100MB groß. Docker build ging wirklich schnell auf meinem Raspberry (insbesondere das zweite Image hat nur eine Sekunde gedauert, da ja quasi alle Layer gleich sind).

Das schöne ist nun, dass man auf dem Raspberry lediglich Docker installiert und alle Dependencies im Container laufen (somit wird es leichter spätere Python Updates zu machen, einfach ein neues Container Image).

Hier mal eine kleine Anleitung, und am Ende des Posts der Inhalt der Dockerfiles. Gerne würde ich Sie auch zu diesem Repo hinzufügen wenn gewünscht:

Als erstes sollte der CAN Adapter nach den Best Practices in Open3e Repo einen festen Namen bekommen und automatisch initialisiert werden: https://github.com/open3e/open3e/blob/master/bestPractices/USB%20Adapter%20and%20udev.md

Wenn dies geschehen ist kann man schon den DepictSystem-Container laufen lassen. Er started für DepictSystem durch und beendet sich dann selbst und legt dabei alle erstellten Dateien im Folder /open3e ab. Dies sollte entsprechend gemappt werden. Nachdem der Container sich beendet hat sollte er noch über docker rm container_id entfernt werden.

`sudo docker run -d --name open3edepict --network host -e CLI_ARGS="-c canvx3" -v ./data:/open3e myimages/open3e-depict:0.1`

Kurze Erläuterung zum docker run Befehl:

- Network Host wird leider benötigt, da das CAN-Interface als Network aufgerufen wird
- Wenn der Can-Adapter anders als can0 gemappt wurde, muss es über die CLI_ARGS übergeben werden (Im Beispiel canvx3).
- Ein lokales Volume (hier ./data) muss auf /open3e gemappt werden. Dort werden dann die Open3e Datapoints und das Devices JSON abgelegt und sollen ja verfügbar bleiben.
- Das Image müsst ihr so nennen wie ihr es im build command genannt habt.

Für open3e verwende ich dann ein docker-compose yaml. es ist wirklich sehr einfach gehalten (das Logging beschränke ich immer noch aus Raspberry SD Karten zeiten :) ). Hier wird nur das Data Folder (oder welches ihr halt auswählt) gemappt. In diesem Folder müssen dann vom depictSystem Container die Dateien liegen und ein `args` File, welches alle Start argumente für open3e enthält.

Das schöne ist an dem Setup, wenn ihr mal einen falschen MQTT Befehl senden und open3e crashed, started es direkt neu (ohne komplexere Daemon Konfiguration.

```
version: "3.8"
services:
  open3e:
    container_name: open3e
    image: "myimage/open3e:0.1"
	network_mode: "host"
    volumes:
      - "./data/:/open3e/"
    restart: always
    logging:
      driver: "json-file"
      options:
       max-file: "2"
       max-size: "10m"
```

Dockerfile DepictSystem:

```
FROM python:alpine3.19

RUN apk update && apk add --no-cache git
RUN mkdir -p /open3e
WORKDIR /open3e

RUN pip install git+https://github.com/open3e/open3e.git

ENV CLI_ARGS=""

CMD open3e_depictSystem ${CLI_ARGS}
```

Dockerfile open3e:

```
FROM python:alpine3.19

RUN apk update && apk add --no-cache git
RUN mkdir -p /open3e
WORKDIR /open3e

RUN pip install git+https://github.com/open3e/open3e.git

CMD open3e @args
```

Ich freue mich auf Diskussionen und Verbesserungsvorschläge.

**Comment by philippoo66** [comment]
moin!

klingt interessant! hast du das alles schon am laufen und ausgiebig getestet? 

ich hatte mal gehört dass es Einschränkungen beim (multiplen) Zugriff auf CAN gibt, weil das eben als Network läuft. Es geht also z.B. nicht, open3e laufen zu lassen und noch nen candump zu machen? 

Grüsse!

**Comment by flecke-m** [comment]
Es war keine Absicht so spät zu antworten, aber ich wollte es jetzt erstmal ein paar Tage laufen lassen. 

Der open3e Docker Container ist ohne Probleme durchgelaufen. Ich habe parallel noch einen E3onCAN Docker Container aufgesetzt um mein E380-CA1 zu belauschen (ein paar Testläufe und seit heute morgen "produktiv"). Das hat alles ohne Probleme geklappt mit dem einen CAN-BUS Adapter.

Ich denke die Limitierungen auf dem CAN-BUS Adapter sind die gleichen wie wenn man die Programme normal auf dem Host laufen lässt. Durch den Network Mode Host sollte der Zugriff genau gleich gehandelt werden.

Das schöne ist nun dass ich zwei Python Apps laufen hab, die nun theoretisch unterschiedliche Dependencies von der Versionierung her haben können, ohne dass es zu Problemen oder komplizierten Setups direkt auf dem Raspberry kommt.

Ich werde nun mal die Datenflüsse zu Home Assistant beobachten, momentan sieht es so aus, als ob die ViCare App teilweise ganz andere Daten hat, wobei ich schätze dass die einfach falsch berechnet werden, da ich neben der VX3 noch ein 4kw SMA Wechselrichter habe, der auch über den E380-CA1 geht und somit wohl Viessmann total verwirrt.

Hier zur Info der Dockerfile für E3onCAN:

```
FROM python:alpine3.19

RUN apk update && apk add --no-cache git

RUN mkdir -p /data
WORKDIR /data
RUN git clone https://github.com/MyHomeMyData/E3onCAN.git

WORKDIR /data/E3onCAN

RUN pip install -r requirements.txt

ENV CLI_ARGS=""

CMD python3 E3onCANcollect.py ${CLI_ARGS}
```

**Comment by Philip-Wiege** [comment]
Ich glaube, dass es mit dem Docker Container sogar möglich wäre, Open3E direkt als Home Assistant Addon laufen zu lassen. Manche haben den HomeAssistant Server ja in CAN Reichweite zur Heizung.

**Reply by philippoo66** [reply]
Das kling auf jeden Fall sehr gut und widerlegt ja eigentlich meine bisherige Vorstellung zur nur-einmal-möglichen Nutzung des CAN Bus im Docker. Hast du mal spasseshalber parallel einen candump versucht?

Wir sollten das auf jeden Fall im Repo aufnehmen denke ich. 

**@ alle was meint ihr?!**

ich denke es wäre dann das beste, dich @flecke-m  als Autor mit in's Boot zu holen, wahrscheinlich muss da ja ab und an auch was ge-maintenace-t werden... Bezüglich E3onCAN ist @MyHomeMyData  zuständig.

**Reply by philippoo66** [reply]
ps. es wäre natrlich auch möglich, im Open3e Repo einen deutlichen Verweis auf ein Docker Repo zu machen...

**Reply by flecke-m** [reply]
Ich habe jetzt auch parallel ein candump gemacht. Beide Container sind munter weitergelaufen ohne Probleme. Bezüglich maintenance des Dockerfiles, ja definitiv muss man regelmäßig das Base Image updaten und je nachdem was sich on open3e ändert (z.B. der Wechsel der Installationsmethode war eine solche Änderung, aber auch Argumente / Variablen je nach Umsetzung).

Es freut mich dass der Vorschlag positiv gesehen wird :)

**Reply by philippoo66** [reply]
was wäre denn eher in deinem Sinne? das in open3e unterzubringen (ich denke ~~als Ordner im Repo?~~ vlt besser als Repo in dem Open3e 'Directory'/Organisation) oder willst du dein eigenes Repo unter deinem 'Namen' machen und wir machen einen dicken Verweis?

**Comment by flecke-m** [comment]
Ich fände den Dockerfile am besten hier in einem Unterordner, damit weitere open3e Nutzer es am schnellsten finden. Meistens kommt man ja aus der Viessmann Community hier her und dann noch ein Schritt und noch ein Schritt macht es nur etwas komplizierter.

Ich könnte parallel mir noch ein Docker Hub Konto eröffnen und das Image dahin pushen. Allerdings würde ich es nur für ARM64 (RPI) machen,

**Reply by rsch90** [reply]
Ich fände ein Image für Proxmox Host ebenfalls geil. Zu mal auch viele mittlerweile ihren Homeassistant weg vom RPI zu einer Proxmox VM umziehen.

**Reply by rsch90** [reply]
Noch besser ein Addon für Homeassistant 😍😍

**Reply by philippoo66** [reply]
was ist denn jetzt Porxmox?

ich denke ein eigenes Docker Repo im open3e wäre besser als ein Unterordner, weil wir ja auch irgendwas mit Revisioning vorhaben, und das dann automatisch auf den ggf. noch nicht aktualisierten Docker Unterordner zuschlagen würde. Das ist dann ein Klick mehr dahin und immer noch im Open3e...

_ein Addon für Homeassistant_ ist ja unabhängig davon, oder? Wär natürlich toll!!

**Reply by flecke-m** [reply]
Docker Repo unter open3e hört sich auch gut an. Ich würde dann die Dockerfiles zur Verfügungstellen und die Beschreibung vom Eingangspost als Readme.

Würdest du dann ein Repo aufmachen und ich erstelle einen PR?

Bezüglich Proxmox, macht es kein Sinn es bereit zustellen, einfach ein LXC Container mit Debian base image (oder etwas was einem auch immer liegt) und dann einfach die Installation wie hier im Repo beschrieben durchführen).


Home Assisstant Add-On ist nochmal etwas besonderes, was soll da erreicht werden? Gerade wenn es um Device Mapping und die Home Assistant installation per Image geht wird es sicher nicht so einfach. 

Ich habe für mich einen extra Raspberry für open3e und einen für Home Assistant, wer möchte kann natürlich gerne die Dockerfiles nehmen und etwas bauen. Ansonsten muss ich sagen, ist es schon recht einfach das ganzen in Home Assistant zu integrieren über MQTT und die im Wiki bereit gestellten Sensor Dateien.

**Reply by philippoo66** [reply]
ich hab https://github.com/open3e/open3e-docker gemacht und dich als member zum open3e eingeladen. ich hoffe das funktioniert jetzt alles so, musst mal schauen.

**Reply by philippoo66** [reply]
ach ja, wegen E3onCAN musst du mal auf @MyHomeMyData warten. Gewöhnlich reagiert er recht schnell, grad scheint er irgendwie unterwegs zu sein oder so. aber ich denke, dieser Container wäre vielleicht besser bei ihm untergebracht? wir könnten das natürlich auch noch hier machen, aber da soll Jürgen erstmal was zu sagen...

**Reply by flecke-m** [reply]
Vielen Dank, es scheint zu funktionieren. Sorry für den Spam bei den Pull requests, ich habe schon lange nicht mehr mit Github gearbeitet.

Wegen E3onCAN schaue ich einfach mal ob @MyHomeMyData interesse hat und dann sollte es auch dort plaziert werden. Ich kann nur sagen dass im Bereich der VX3 und des Energiezählers E380-CA1 open3e und E3onCAN gute Partner sind.

**Reply by philippoo66** [reply]
sehr schön, vielen Dank! ich dachte, du könntest das jetzt 'eigenmächtig' hochladen, wäre auch für maintenance ganz gut. aber vlt ist ganz open3e nach 4-Augen Prinzip angelegt, ich hab da doch auch keine Ahnung von... ;-)

**Reply by philippoo66** [reply]
fehlt da nicht noch was? die images oder so? ich dachte in einem docker wäre 'alles drin' (sorry, ich hab eben so überhaupt keine Ahnung...)

**Reply by flecke-m** [reply]
Mit dem Dockerfile kann man sich das image über den docker build Befehl selbst bauen.

Ich werde schauen ob ich ein Docker Hub Account erstelle und dort meine Images hochlade.

**Reply by philippoo66** [reply]
erklär doch bitte mal (ich möchte ja nicht ganz so ahnungslos sterben irgendwann...) warum kann man so ein image nicht hier hochladen?

**Reply by flecke-m** [reply]
Da brauchst du leider ein Repository. GitHub hat dafür GitHub Packages aber das kostet relativ schnell etwas. 

Docker Hub ist noch kostenlos für öffentliche Repositories. 

Ich habe mal schnell ein Account erstellt und die Images hochgeladen: https://hub.docker.com/repositories/fleckem

Werde demnächst den Link in die Readme einfügen, heute aber nicht mehr.

**Reply by MyHomeMyData** [reply]
@philippoo66 @flecke-m Hi, mich gibt's auch noch. Bin halt mal wieder im Urlaub. Wir waren die letzten beiden Tage unterwegs, sind jetzt gut in den Pyrenäen angekommen. Und es gibt wieder WLAN 😄 

Gerne nehme ich die Docker-Variante für E3onCAN auf. Arbeite selbst aber kaum mit Docker, ihr müsst mir also erklären, was ich machen soll.

Schöne Grüße aus Frankreich
Jürgen

P.S.: Im französischen Fernsehen werden nur die 21-Uhr-Spiele der EM übertragen. ard.de sendet kein Live-Bild an IP-Adressen außerhalb Deutschlands. Aber ich war vorbereitet. Mit einem VPN nach Deutschland hat's wunderbar funktioniert.

**Reply by philippoo66** [reply]
ich hab mal grad auf den Link geklickt

![grafik](https://github.com/open3e/open3e/assets/122479122/63e7ac6a-190f-4515-9354-b9af94e37cdd)

kommt das weil ich Docker nicht meine Daten geben will oder hast du da noch irgendwas auf privat oder restricted oder so gestellt?

**Reply by philippoo66** [reply]
für die zwei files und das Readme könnte man natürlich doch einen Unterordner in open3e/open3e machen. Ich dachte da kommt halt alles hin inkl. images...

**Reply by philippoo66** [reply]
ps. Schönen Urlaub euch @MyHomeMyData !! :-)

**Comment by flecke-m** [comment]
Ja ein Dockerfile ist nicht mehr, der reicht um das Image zu bauen.

Da hab ich leider falsch verlinkt.

Der sollte klappen: https://hub.docker.com/u/fleckem

@MyHomeMyData viel Spaß im Urlaub, du kannst mich ja danach kontaktieren.

**Reply by philippoo66** [reply]
wollen wir dann die Files doch in einen Unterordner von open3e verschieben und das neue Repo löschen?

die sehen ja so auch erstmal ziemlich versionsunabhängig aus...

**Reply by MyHomeMyData** [reply]
@flecke-m Also ich kann hier schon was machen. Heute hat es abwechselnd geregnet und geschüttet ...

Wenn ich es richtig sehe, sollte ich ein Repo `E3onCAN-docker` anlegen. Dann erzeugst Du einen PR, der Readme und Docker-File anlegt. Richtig?

Gibt es besondere Anforderungen an das Repo?

**Reply by flecke-m** [reply]
@philippoo66 ja das können wir machen es sollte eigentlich gehen.

@MyHomeMyData ich denke wie hier besprochen würde ein Unterordner "Docker" wohl am besten passen. Es ist nur ein File und eine Readme.

**Reply by MyHomeMyData** [reply]
@flecke-m Ordner "Docker" und ein leeres README hab ich im Branch develop angelegt. Erzeugst Du einen PR?

**Reply by flecke-m** [reply]
@MyHomeMyData 

Ich habe den PR für den develop Branch erstellt. Ich habe diesmal auch direkt auf Docker Hub verwiesen, wenn jemand es auf einem Raspberry Pi 4 laufen lässt.

**Reply by philippoo66** [reply]
> wollen wir dann die Files doch in einen Unterordner von open3e verschieben und das neue Repo löschen?

@MyHomeMyData Wie machen wir das? gleich im master? oder besser im develop? ist das eigentlich aktuell synchron?

**Reply by MyHomeMyData** [reply]
@flecke-m Super, hab ich gemerged und einen Hinweis ins E3onCAN-Readme eingefügt. Bitte schaue mal kurz drüber. Wenn es passt, merge ich nach main.

**Reply by MyHomeMyData** [reply]
@philippoo66
> Wie machen wir das? gleich im master? oder besser im develop? ist das eigentlich aktuell synchron?

Im open3e develop branch sind 2 commits von Dir dazu gekommen, betrifft nur `src/open3e/Open3E_depictSystem.py` soweit ich das sehe.

Wenn das nach master soll, wäre es am besten, die Docker-Änderungen in develop einzubauen und dann alles nach master zu mergen.

**Reply by philippoo66** [reply]
ok, erledigt. Ordner docker ist im develop. Machst du bitte den Rest!?

**Reply by MyHomeMyData** [reply]
@philippoo66 @flecke-m Merge develop -> master erledigt. Wäre super, wenn ihrs kurz ausprobieren könntet. Bin da im Moment ein wenig eingeschränkt.

**Reply by flecke-m** [reply]
@MyHomeMyData @philippoo66 Ich habe es mir angeschaut sieht super aus.

**Reply by MyHomeMyData** [reply]
@flecke-m Ist in main verfügbar. Nochmals danke für den super Beitrag!

**Reply by wesch** [reply]
Hallo,
Ich habe noch ein paar Änderungen für eine Vitocal250AH gemacht. Falls Interesse besteht, hier sind sie. Ich wollte sie nicht selbst einpflegen, da ich nicht weiß, ob es für euch ok ist. 

543 : O3EComplexType(4, "SmartGridReadyConsolidator",[O3EByteVal(1, "unknown"), O3EByteVal(1, "OperatingStatus"), O3EByteVal(1, "unknown"),O3EByteVal(1, "unknown")]),

2543 : O3EComplexType(10, "SmartGridTemperatureOffsets",[RawCodec(2,"Unknown"),O3EInt16(2, "SetTempRoom", scale=10.0), O3EInt16(2, "SetTempDomesticHotWater", scale=10.0),O3EInt16(2, "SetTempBuffer", scale=10.0)]),

2940 : O3EComplexType(3, "ExternalHeaterDelayTimer",[O3EInt8(1, "SwitchOnDelay", scale=1.0), O3EInt8(1, "MinRunTime", scale=1.0), O3EInt8(1, "SwitchOffDelay", scale=1.0)]),

3106 : O3EComplexType(4, "BufferMinimumMaximumSetTemperature",[O3EInt16(2, "BufferMin", scale=10.0, signed=True), O3EInt16(2, "BufferMax", scale=10.0, signed=True)]),

Viele Grüße

**Comment by wesch** [comment]
Hallo,
erst einmal super vielen Dank für die Docker Anleitung. Es funktioniert großartig. Ich hatte nur ein kleines Problem mit meiner eigenen Open3Edatapoints.py. Ich habe noch ein paar modifikationen gemacht und sie haben in der Dockerversion einfach nicht funktioniert. 
Die Lösung:
Open3Edatapoints.py muss nach /usr/local/lib/python3.12/site-packages/open3e/Open3Edatapoints.py kopiert werden. 

Meine Änderungen betreffen eine Vitocal 250 AH. Ich wollte die Änderungen nicht einfach ins Repo stellen. 

Wer Interesse hat:

543 : O3EComplexType(4, "SmartGridReadyConsolidator",[O3EByteVal(1, "unknown"), O3EByteVal(1, "OperatingStatus"), O3EByteVal(1, "unknown"),O3EByteVal(1, "unknown")]),

2543 : O3EComplexType(10, "SmartGridTemperatureOffsets",[RawCodec(2,"Unknown"),O3EInt16(2, "SetTempRoom", scale=10.0), O3EInt16(2, "SetTempDomesticHotWater", scale=10.0),O3EInt16(2, "SetTempBuffer", scale=10.0)]),

2940 : O3EComplexType(3, "ExternalHeaterDelayTimer",[O3EInt8(1, "SwitchOnDelay", scale=1.0), O3EInt8(1, "MinRunTime", scale=1.0), O3EInt8(1, "SwitchOffDelay", scale=1.0)]),
       
3106 : O3EComplexType(4, "BufferMinimumMaximumSetTemperature",[O3EInt16(2, "BufferMin", scale=10.0, signed=True), O3EInt16(2, "BufferMax", scale=10.0, signed=True)]),
       

Viele Grüße

**Reply by philippoo66** [reply]
danke @wesch ! ich habe das im aktuellen develop eingearbeitet.
Grüsse!
Phil

**Comment by wesch** [comment]
noch eine Bemerkung: die Kopie muss innerhalb des Containers durchgeführt werden. D.h. man attached sich in den Container und führt dort die Kopie nach /usr/local/lib/python3.12/site-packages/open3e.

Geht am besten mit Portainer. (Achtung in Alpine gibt es kein bash sondern nur /bin/sh)

**Reply by flecke-m** [reply]
Danke für dein Feedback 👍 

Dein Vorschlag zur Anpassung der Open3Edatapoints.py müsstest du an @philippoo66 oder @MyHomeMyData richten, da dies im Source Code von Open3e gemacht werden muss.

Ein kleiner Hinweis zu deiner Methodik. Falls du den Container mal komplett ne erstellst müsstest du dein Open3Edatapoints.py wieder kopieren.

Du kannst auch versuchen, dass du über Docker-Volume mount den File von deinem host in /usr/local/python3.12/site-packages/open3e/Open3Edatapoints.py mountest, dann hast du immer deine Version im Container. 

Generell würde ich es aber bevorzugen, wenn die Änderung (falls Sie für alle Nutzer relevant sein kann) in open3e kommt und ich dann einfach ein neues Image baue und nach Docker Hub pushe.

**Reply by philippoo66** [reply]
> Dein Vorschlag zur Anpassung der Open3Edatapoints.py müsstest du an @philippoo66 oder @MyHomeMyData richten, da dies im Source Code von Open3e gemacht werden muss.

@flecke-m eigentlich solltest du auch berechtigt sein, Änderugen vorzunehmen?

**Reply by flecke-m** [reply]
Ja ich kann Änderungen vornehmen, da ich aber ein absoluter Python Beginner bin und noch nicht ganz verstanden habe wie euer Code funktioniert, nehme ich davon erstmal abstand, da ich immer wissen will was ich tue^^

Momentan beschränke ich mich deshalb erstmal auf Docker. Wenn ich meine Vitoair angeschlossen habe, werde ich dann evtl. auch mal tiefer einsteigen, da diese ja anscheinend noch nicht so gut dokumentiert ist. Das dauert aber noch.

**Comment by wesch** [comment]
Danke, sehr guter Vorschlag. Ich versuche das zu Mappen. Dann wird es auch einfacher, falls man einen Fehler macht. Die Änderungen gebe ich an die Kollegen weiter. Am 26.06.2024 um 09:51 schrieb flecke ***@***.***>:﻿
Danke für dein Feedback 👍
Dein Vorschlag zur Anpassung der Open3Edatapoints.py müsstest du an @philippoo66 oder @MyHomeMyData richten, da dies im Source Code von Open3e gemacht werden muss.
Ein kleiner Hinweis zu deiner Methodik. Falls du den Container mal komplett ne erstellst müsstest du dein Open3Edatapoints.py wieder kopieren.
Du kannst auch versuchen, dass du über Docker-Volume mount den File von deinem host in /usr/local/python3.12/site-packages/open3e/Open3Edatapoints.py mountest, dann hast du immer deine Version im Container.
Generell würde ich es aber bevorzugen, wenn die Änderung (falls Sie für alle Nutzer relevant sein kann) in open3e kommt und ich dann einfach ein neues Image baue und nach Docker Hub pushe.

**Comment by philippoo66** [comment]
@wesch "Run failed: Test - develop: FAILED tests/test_codecs.py::test_datapoints_length - AssertionError: Did 2543: Length 10 does not match sum of subType lenghts 8"

da hat es Recht. Weisst du, was da fehlt bzw wo was fehlt? ein 'unknown' würde ja reichen...

**Reply by wesch** [reply]
Sorry, ich bin auch kein Python Profi. Die Meldung kam bei mir nicht.  Es muss noch ein unknown dahinter.

**Reply by philippoo66** [reply]
die Meldung kommt aus den automatisierten Tests. die scheinen ja Sinn zu machen ;-) 
ich häng noch 2 Bytes unknown dahinter.

**Reply by philippoo66** [reply]
![grafik](https://github.com/open3e/open3e/assets/122479122/cc46b689-71e9-4533-9aab-cae0bcd0a926)

:-)

**Comment by wesch** [comment]
Danke, perfekt. Am 27.06.2024 um 11:17 schrieb Phil ***@***.***>:﻿
die Meldung kommt aus den automatisierten Tests. die scheinen ja Sinn zu machen ;-)
ich häng noch 2 Bytes unknown dahinter.

**Reply by SanMiggel** [reply]
Ha freut mich, dass die Tests zu was nutze sind. 😀

Die Tests laufen auch bei einem pull request nach Develop oder master. Also Feature branch und dann pull request wäre der bessere work flow. 😉

**Comment by flecke-m** [comment]
Nachdem ja die neuesten Änderungen per Pull-Request in Master gemerged wurden, habe ich die images open3e und open3e-depict auf v0.2 geupdated. Zusätzlich habe ich das Base Alpine Image 3.20 statt 3.19 genommen.

Depict habe ich noch nicht getestet, aber das open3e läuft mit meiner VX3 und E380 einwandfrei.

**Comment by TheSmartGerman** [comment]
Hallo,
mit ist aufgefallen, readme sollte "fleckem" noch druch "user" ersetzt werden, oder?

https://github.com/open3e/open3e/blob/master/docker/README.md

**Reply by philippoo66** [reply]
gute Idee! :-)

**Reply by flecke-m** [reply]
Das geht leider nicht. Ich habe mir ein Account bei Docker Hub gestellt und so kann man das Image da nur herunterladen.

Es gibt leider keine Möglichkeit kostenlose Organisations Repos mehr zu erstellen. Umbenennen kann ich mich auch nicht mehr...

**Reply by philippoo66** [reply]
alles klar. dann bleibt's halt so. So einen Zusammenhang hatte ich nicht auf dem Schirm...

**Reply by TheSmartGerman** [reply]
Ok, ja bin auch nicht so Tief im docker drin. Möchte da die nächsten Zeit einige Updates an meinen system fahren und sondier was optimal ist... Und bei den Pfadangaben bon ich stutzig geworden.

**Comment by flecke-m** [comment]
So ich hatte beim aktualisieren des Docker Images leider ein paar Probleme. Mit dem neuesten Python Base Image (Python 3.13.0) hat es leider ein Fehler im Build Prozess gegeben (g++ war nicht installiert), wenn ich g++ manuell zu erst installiert habe war das Image 3 Mal so groß wie vorher... 

Ich habe jetzt die Docker Images (open3e und open3e-depict) mit dem Alpine Image mit Python 3.12.7 gebaut. Zwischendurch hatte ich dann noch ein paar Probleme mit dem build-cache. Aber nun läuft mein open3e mit der Version v0.2.1 (bitte benutzt die gerade hoch geladene Version, ich habe sie aus den oben genannten Gründen nochmal überschreiben müssen). 

Ebenso verwende ich das Versioning hier aus dem Repo. Das sieht erstmal blöd aus, da ich schon bei v0.3 war aber einfach im Docker Hub das neueste Image nehmen (v0.2.1). Ich lösche die älteren Images aus Kompatibilitätsgründen nicht.

**Comment by lgraf** [comment]
@flecke-m ich habe eine [Diskussion](https://github.com/open3e/open3e/discussions/196) zum Thema automatisieren von Docker Image builds gestartet. Ich vermute, du hast sicherlich auch Interesse an dem Thema. Falls du Zeit hast, schau gerne mal drauf, würde mich über Feedback/Input von deiner Seite freuen.

**Reply by philippoo66** [reply]
@flecke-m wir planen ein Entwicklertreffen. https://github.com/open3e/open3e/discussions/199 Was is mit dir?!?

**Reply by flecke-m** [reply]
Danke dass ihr mich direkt angesprochen habt, dadurch habe ich die Diskussionen jetzt gesehen. Ich werde in beiden Themen mein Kommentar schreiben. Ich wollte auch die Version 0.5.0 endlich wieder als Docker image publishen, wobei ich mein build prozess ändern werden und versuche mit buildx ein Multi-Arch Image bereit zustellen, so dass nicht nur Raspberry Image, sondern auch ein x86 image zur Verfügung steht. Ich glaube erstmal mache schnell (noch diese Woche) das Raspberry Image und dann schau ich mal wie es klappt.

**Reply by philippoo66** [reply]
Kommentar schreiben ist schön und gut, aber ich glaube es wäre auch sinnvoll, wenn du bei dem Gespräch (voraussichtlich geplant Di abend 18.2. 21h) dabei wärst!?

**Comment by flecke-m** [comment]
So mal ein Update. Ich habe mal meine Docker build VM angeschmissen und einen ersten Multi-Arch build gemacht (Arm64 und Amd64). Das Arm64 image läuft bei mir jetzt auf dem Raspi mit meiner VX3 ohne Probleme. 

Viel Spass mit der aktuellen Version (ich habe auch e3oncan geupdated, für alle die es benutzen).

**Comment by flecke-m** [comment]
Hallo,

ich habe mal endlich einen Wiki-Eintrag für die Docker Images erstellt und ebenso mein neues Docker Image für noch einfachere Verwendung referenziert. [Wiki](https://github.com/open3e/open3e/wiki/033-Docker-and-Home-Assistant-Addon)

Das Home Asssistant Add-On wird dort auch direkt beschrieben. Ich hoffe es hilft weiter und gerne Feedback geben, falls ihr Verbesserungsvorschläge habt.

**Reply by philippoo66** [reply]
ich kann leider zu dem ganzen Docker Krams nix sagen, weil ich weder irgendeine Ahnung davon habe noch es benutze noch open3e überhaupt benutze (mangels Zielsystem)

aber vielen Dank für deine Arbeit!! die Dokumentation zu machen ist ja doch immer 'ungeliebt'....  ;-)
